ksql-engine/src/main/java/io/confluent/ksql/util/EngineProcessingLogMessageFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.json.JsonMapper;
+import io.confluent.ksql.processing.log.ProcessingLogMessageSchema;
+import io.confluent.ksql.processing.log.ProcessingLogMessageSchema.MessageType;
+import java.util.function.Supplier;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.data.Struct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class EngineProcessingLogMessageFactory {
+  private static final Logger LOGGER
+      = LoggerFactory.getLogger(EngineProcessingLogMessageFactory.class);
+
+  private EngineProcessingLogMessageFactory() {
+  }
+
+  public static Supplier<SchemaAndValue> recordProcessingError(
```

Comment:
I think we should be more diligent about adding javadocs so that newbies like me can ramp up faster and more independently :) 

Also personal preference nit (feel free to ignore), instead of having this return a supplier you can have it return a `SchemaAndValue` and then reference it as `Engine..Factory::recordProcessingError` to reduce some bloat. Alternatively, if you plan to keep this as the only method in the class, the class can just implement `Supplier<SAV>`

Reply:
Normally I'd agree. I'm going to leave it as is just because the cost of accidentally calling the factory function rather than just passing it is pretty high.